distributed/client.py

```diff
@@ -3035,22 +3035,11 @@ def restart(self, **kwargs):
         return self.sync(self._restart, **kwargs)
 
     async def _upload_file(self, filename, raise_on_error=True):
-        with open(filename, "rb") as f:
-            data = f.read()
-        _, fn = os.path.split(filename)
-        d = await self.scheduler.broadcast(
-            msg={"op": "upload_file", "filename": fn, "data": to_serialize(data)}
+        await self._register_worker_plugin(
+            UploadFile(filename),
+            name=filename + str(uuid.uuid4()),
```

Comment:
Are we making the `name` here unique so that if a file is updated locally and then re-uploaded that the new version of the file will be present on the workers? If so, could you add a test for this case (if one doesn't already exist)

Reply:
Yes, that's right. Without this unique name, re-uploading the same file with changes has no effect. This is already tested by existing tests, which do exactly that.

Reply:
Great!